### ISSUE TITLE
Tree.Expand delegate that is async

### ISSUE THREAD

**Issue by duanepfeiffer** [radzenhq/radzen-blazor#530]
**Describe the bug**
If your Expand delegate has an await and one of your tree items is expanded using Expanded delegate (ie expanded via code), children are not displayed.

**To Reproduce**
Make the following changes to the Tree->Files and directories demo
**_markup_**: 
                <RadzenTreeLevel Text=@GetTextForNode Template=@FileOrFolderTemplate
                    Expanded=@(n => Path.GetFileName((string)n) == "runtimes")/>
**_code_**:
1. Make LoadFiles async Task
2. Add await Task.Delay(2000)

**Expected behavior**
When app displays, the files in the runtimes folder should show.

**Screenshots**
Get:
![image](https://user-images.githubusercontent.com/22204576/178831557-ed717a07-def0-4771-84f3-1225df66a4d0.png)

Should Get:
![image](https://user-images.githubusercontent.com/22204576/178831642-ab3bff9b-1bf6-467c-8af5-367cdb11a08d.png)

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser Chrome
 - Version 3.19.4

[Index.zip](https://github.com/radzenhq/radzen-blazor/files/9105950/Index.zip)

**Comment by duanepfeiffer** [comment]
Not sure if this is the correct way to fix this but for now I've added StateHasChanged to the bottom of ExpandItem in RadzenTree.razor.cs

        internal async Task ExpandItem(RadzenTreeItem item)
        {
            var args = new TreeExpandEventArgs()
            {
                Text = item?.Text,
                Value = item?.Value,
                Children = new TreeItemSettings()
            };

            await Expand.InvokeAsync(args);

            if (args.Children.Data != null)
            {
                Debug.WriteLine("<<RADZEN>> Rendering children...");

                var childContent = new RenderFragment(builder =>
                {
                    Func<object, string> text = null;
                    var children = args.Children;

                    foreach (var data in children.Data)
                    {
                        if (text == null)
                        {
                            text = children.Text ?? Getter<string>(data, children.TextProperty);
                        }

                        RenderTreeItem(builder, data, children.Template, text, children.HasChildren, children.Expanded, children.Selected);
                        builder.CloseComponent();
                    }
                });

                item.RenderChildContent(childContent);

                if (AllowCheckBoxes && AllowCheckChildren && args.Children.Data != null)
                {
                    if (CheckedValues != null && CheckedValues.Contains(item.Value))
                    {
                        await SetCheckedValues(CheckedValues.Union(args.Children.Data.Cast<object>().Except(UncheckedValues)));
                    }
                    else
                    {
                        await SetCheckedValues(CheckedValues.Except(args.Children.Data.Cast<object>()));
                    }
                }
            }
            else if (item.Data != null)
            {
                if (AllowCheckBoxes && AllowCheckChildren && item.Data != null)
                {
                    if (CheckedValues != null && CheckedValues.Contains(item.Value))
                    {
                        await SetCheckedValues(CheckedValues.Union(item.Data.Cast<object>().Except(UncheckedValues)));
                    }
                    else
                    {
                        await SetCheckedValues(CheckedValues);
                    }
                }
            }
            **this.StateHasChanged();**
        }

**Comment by akorchev** [comment]
This is what happens with this code:

```
    async Task LoadFiles(TreeExpandEventArgs args)
    {
        var directory = args.Value as string;

        await Task.Delay(2000);
        args.Children.Data = Directory.EnumerateFileSystemEntries(directory);
        args.Children.Text = GetTextForNode;
        args.Children.HasChildren = (path) => Directory.Exists((string)path);
        args.Children.Template = FileOrFolderTemplate;
    }
```
![tree](https://user-images.githubusercontent.com/454726/178966182-0f61673c-3d4d-4b35-b732-802530f8c0be.gif)

The nodes appear after a delay without having an extra StateHasChanged() call.

**Comment by duanepfeiffer** [comment]
Your example is showing user click to expand?  That is not the way to reproduce the problem.  Did you change your html markup as described in my post?

<RadzenTreeLevel Text=@GetTextForNode Template=@FileOrFolderTemplate
**Expanded=@(n => Path.GetFileName((string)n) == "runtimes")**/>

**Comment by akorchev** [comment]
I see. Should be fixed by d6760ca4b9ddc7ea42f8bb830a0263e58b048cf7
